tests/test_util.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+from pytest_mock import mocker
+from src.cisco_gnmi.proto import gnmi_pb2
+from src.cisco_gnmi import util
+from src.cisco_gnmi.util import urlparse, ssl, x509, default_backend
+
+
+def test_gen_target_netloc_valueerror():
+    with pytest.raises(ValueError):
+        util.gen_target_netloc("http://www.test.com", "test_prefix://", 8000)
+
+def test_gen_target_netloc_parsed_none():
+
+    mock_netloc = '//www.testing.com:8080'
+    mock_parsed_netloc = urlparse(mock_netloc)
+
+    result = util.gen_target_netloc("www.testing.com", "http://", 8080)
+    assert mock_parsed_netloc == result
+
+def test_validate_proto_enum_exception_one():
+
+    enum = gnmi_pb2.SubscriptionMode
+
+    with pytest.raises(Exception):
+        util.validate_proto_enum('test', 'INVALID_VALUE', 'test', enum)
+
+def test_validate_proto_enum_exception_two():
+
+    enum = gnmi_pb2.SubscriptionMode
+    fake_subset = [3]
+
+    with pytest.raises(Exception):
+        util.validate_proto_enum('test', 2, 'test', enum, subset=fake_subset)
+
+def test_validate_proto_enum_exception_three():
```

Comment:
Might be able to make the methods a little more specific.
`test_validate_proto_enum_exception_subset`? Or is common practice to enumerate `_one`, `_two`, ...?

Reply:
It is not common to use `_one`, '_two` but it is better since we can identify to which exception in the method we are referring to, etc but it has to be done like that since you don't have a specific exception. you have some general exceptions there.
You have to create your own exceptions.